packages/strapi-plugin-upload/admin/src/utils/createNewFilesToDownloadArray.js

```diff
@@ -29,7 +29,7 @@ const createNewFilesToDownloadArray = (filesURLArray, alreadyUploadedFiles) => {
       fileInfo: {
         alternativeText: '',
         caption: '',
-        name: current,
+        name: current.substring(current.lastIndexOf('/') + 1),
```

Comment:
I think this won't be enough here (we need to handle query strings :/)

We should use URL to parse the url get the name:

```js
try {
        const url = new URL(current);

        const name =  decodeURIComponent(url.pathname.substring(url.pathname.lastIndexOf('/') + 1))

 } catch (err) {
  // invalid url
} 
```

Reply:
Ok I'll make an update then